### DISCUSSION TITLE
flexibly change trial type within a sequence.

### DISCUSSION THREAD

**Discussion by HuangHam** [jspsych/jsPsych#2762] [Q&A] [answered]
Hi, I have the following sequence:
var BDM_sequence = {
        timeline: [BDM, fixation, realize],
        timeline_variables: BDM_cond,
        randomize_order: true
      };

I would like to vary the type of the component "realize" depending on the subject's response during the component "BDM". For example, if the subject's response is correct in BDM, "realize" would be a text component; if it's incorrect, "realize" would be an image component. Can anyone help me with this, please?

**Comment by jodeleeuw** [comment, marked as answer]
You can make `realize` a [conditional timeline](https://www.jspsych.org/7.3/overview/timeline/#conditional-timelines) that is nested in `BDM_sequence`. You may need two conditional timelines, one for the text component and one for the image component.
